Main/src/main/java/com/main/acad/util/ConnectionPool.java

```diff
@@ -1,37 +1,82 @@
 package com.main.acad.util;
 
-import org.apache.commons.dbcp.BasicDataSource;
-
-import javax.sql.DataSource;
 import java.io.IOException;
 import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
 import java.util.Properties;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.logging.Logger;
 
 public class ConnectionPool {
 
-    private static BasicDataSource dataSource;
+    private static ConnectionPool instance;
+    private static final Logger logger = Logger.getLogger(ConnectionPool.class.getName());
+    private BlockingQueue<Connection> pool;
+    private int maxPoolSize;
+    private int currentPoolSize;
+    private String dataBaseUrl;
+    private String dataBaseUser;
+    private String dataBasePassword;
+    private static ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+    private static InputStream inputStream = classLoader.getResourceAsStream("config.properties");
+    private static Properties properties = new Properties();
 
-    static ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-    static InputStream input = classLoader.getResourceAsStream("config.properties");
-    static Properties prop = new Properties();
+    private ConnectionPool() {
+        this.maxPoolSize = 100;
+        this.pool = new LinkedBlockingQueue<>(maxPoolSize);
+        this.currentPoolSize = 0;
+        this.dataBaseUrl = properties.getProperty("url");
+        this.dataBaseUser = properties.getProperty("login");
+        this.dataBasePassword = properties.getProperty("password");
+        try {
+            Class.forName(properties.getProperty("driver"));
+            openAndPoolConnection();
+        } catch (ClassNotFoundException e) {
+            logger.info("An error occurred in ConnectionPool class with private Constructore");
```

Comment:
We don't use application after this error, so you should rethrow another unchecked

Reply:
What did you mean?
throw my exception?

Reply:
This comment is also about exception handling, do the same as for others catch block - rethrow unchecked